test/Parse/recovery.swift

```diff
@@ -435,8 +435,8 @@ struct ErrorTypeInVarDeclFunctionType1 {
   var v2 : Int
 }
 
-struct ErrorTypeInVarDeclArrayType1 { // expected-note{{in declaration of 'ErrorTypeInVarDeclArrayType1'}}
-  var v1 : Int[+] // expected-error {{expected declaration}} expected-error {{consecutive declarations on a line must be separated by ';'}}
+struct ErrorTypeInVarDeclArrayType1 {
+  var v1 : Int[+] // expected-error {{unexpected ']' in type; did you mean to write an array type?}}
```

Comment:
Interesting. I don't actually think we should give this message here; do you know why it's happening?

Reply:
There is some really old code here that is trying to teach you not to use C-style arrays. (`new_array_syntax` with the diagnostic "array types are now written with the brackets around the element type")

So it parses `Int`, thinks `[` is the beginning of a C-style array declaration, tries to parse a size expression, sees that `+` by itself isn't a valid expression, gives up (but has now consumed the `[+`, and then comes back out to hit this new diagnosis on the `]`.

My feeling is that this example is so weird, that any sort of error message with `unexpected` in it is as good as it was before.